crates/proof-of-sql-planner/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+//! This crate converts a `DataFusion` `LogicalPlan` to a `ProofPlan` and `Postprocessing`
```

Comment:
This would allow you to remove the `allow`s everywhere else, and makes it easier to deal with in the future.
```suggestion
//! This crate converts a `DataFusion` `LogicalPlan` to a `ProofPlan` and `Postprocessing`
#![cfg_attr(not(test), expect(dead_code))] // TODO: remove this when initial development work is done
#![cfg_attr(test, expect(clippy::missing_panics_doc))]
```